osquery/tables/system/windows/windows_security_products.cpp

```diff
@@ -0,0 +1,203 @@
+#include <codecvt>
+#include <locale>
+
+#include <iwscapi.h>
+#include <wscapi.h>
+
+#include <osquery/core.h>
+#include <osquery/core/windows/wmi.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/utils/map_take.h>
+#include <osquery/utils/scope_guard.h>
+
+namespace osquery {
+namespace tables {
+
+const auto kSecurityProviders = std::unordered_map<int, std::string>{
+    {WSC_SECURITY_PROVIDER_FIREWALL, "Firewall"},
+    {WSC_SECURITY_PROVIDER_ANTIVIRUS, "Antivirus"},
+    {WSC_SECURITY_PROVIDER_ANTISPYWARE, "Antispyware"},
+};
+
+const auto kSecurityProviderStates = std::unordered_map<int, std::string>{
+    {WSC_SECURITY_PRODUCT_STATE_ON, "On"},
+    {WSC_SECURITY_PRODUCT_STATE_OFF, "Off"},
+    {WSC_SECURITY_PRODUCT_STATE_SNOOZED, "Snoozed"},
+    {WSC_SECURITY_PRODUCT_STATE_EXPIRED, "Expired"},
+};
+
+struct wsc_entry {
+  WSC_SECURITY_PROVIDER provider;
+  std::wstring product_name;
+  WSC_SECURITY_PRODUCT_STATE product_state;
+  std::wstring product_state_timestamp;
+  std::wstring remediation_path;
+  WSC_SECURITY_SIGNATURE_STATUS signature_status;
+};
+
+Status GetSecurityProducts(WSC_SECURITY_PROVIDER provider,
+                           std::vector<wsc_entry>& out_list) {
+  // Much of the following is adapted from the MS example at
+  // https://github.com/Microsoft/Windows-classic-samples/blob/master/Samples/WebSecurityCenter/cpp/WscApiSample.cpp
+
+  HRESULT hr = S_OK;
+  IWscProduct* PtrProduct = nullptr;
+  IWSCProductList* PtrProductList = nullptr;
+  BSTR PtrVal = nullptr;
+  LONG ProductCount = 0;
+  WSC_SECURITY_PRODUCT_STATE ProductState;
+  WSC_SECURITY_SIGNATURE_STATUS SignatureStatus;
+
+  const auto guard =
+      scope_guard::create([&PtrProduct, &PtrProductList, &PtrVal]() {
+        // Be sure to clean up any lingering pointers before return
+        if (nullptr != PtrVal) {
+          SysFreeString(PtrVal);
+        }
+        if (nullptr != PtrProductList) {
+          PtrProductList->Release();
+        }
+        if (nullptr != PtrProduct) {
+          PtrProduct->Release();
+        }
+      });
+
+  if (provider != WSC_SECURITY_PROVIDER_FIREWALL &&
+      provider != WSC_SECURITY_PROVIDER_ANTIVIRUS &&
+      provider != WSC_SECURITY_PROVIDER_ANTISPYWARE) {
+    VLOG(1) << "Invalid security provider code: " << provider;
+
+    return Status(E_INVALIDARG);
```

Comment:
There are actually only 2 different codes supported by `Status` 0 and 1. So, I'd suggest to use `Status::failure()` and `Status::success()` instead.